client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
+
+#### Code example
 
 ```tsx
 /**
- * This is the return type of useStepNavigation hook
+ * Each step must be typed as Step. And it should declare the types of the data it submits and the data it accepts.
  */
-export type NavigationControls = {
+const SelectImportedSiteSource: Step< {
+	// This steps submits `platform` and `url` strings.
+	submits: {
+		platform: 'Wix' | 'Squarespace';
+		url: string;
+	};
+	// And it accepts the following props. Adding them here will make them available in the `props` object.
+	accepts: {
+		title?: string;
+		subTitle?: string;
+	};
+} > = function ImportStep( props ) {
+	const siteSlug = useSiteSlug();
 	/**
-	 * Submits the answers provided in the flow
+	 * `navigation` prop is provided by Stepper. It allows your step to submit.
 	 */
-	submit?: ( providedDependencies?: ProvidedDependencies, ...params: string[] ) => void;
+	const { navigation, title, subTitle } = props;
+
+	return (
+		<div>
+			<h1>{ title || 'Welcome to WordPress.com' }</h1>
+			<h2>{ subTitle || 'Time to migrate you a site!' }</h2>
+
+			<form
+				onSubmit={ ( event ) =>
+					navigation.submit( {
+						platform: event.currentTarget.platform,
+						url: event.currentTarget.url,
+					} )
+				}
+			>
+				<label>
+					Enter the URL of your site
+					<input type="url" name="url" />
+				</label>
+
+				<label>
+					Pick your current site platform
+					<select name="platform">
+						<option value="wix">Wix</option>
+						<option value="ss">Squarespace</option>
+					</select>
+				</label>
+
+				<button>Submit</button>
+			</form>
+		</div>
+	);
 };
+
+export default SelectImportedSiteSource;
 ```
 
-Since this is a hook, it can access any state from any store, so you can make dynamic navigation decisions based on the state. [Here](./declarative-flow/site-setup-flow.ts) is a developed example of this hook.
+#### Passing data down to steps
+
+It is often the case that you want to customize steps around your flow. You can do that via `useStepsProps`.
+
+#### Reusability
+
+Stepper aims to create a big `steps-repository` that contains the steps and allows them to be recycled and reused. Every step you create is inherently reusable by any future flow. Because steps are like components, they're not parts of the flows, flows just happen to use them.
+
+**This creates a couple of restrictions.**
+
+1. Flow-specific styling should be done in a `style.scss` file put in the flow's folder. Each step should have the basic styling necessary to operate on its own.
+2. Steps should not do `if ( flow === 'X' ) do Y`. This is a very common pattern. It was a necessary evil before we introduced `useStepsProps`. But now, it's an unnecessary evil 😬
+
+#### Renaming steps
+
+There may be a time when a step needs to be renamed. In order to preserve Tracks data and funnels, we recommend adding a new entry to `getStepOldSlug` in the `FlowRenderer` component. This ensures that tracks events will fire with both the new step slug and the old step slug.
```

Comment:
I have found [a few uses](https://github.com/search?q=repo%3AAutomattic%2Fwp-calypso%20getStepOldSlug&type=code) of `getStepOldSlug` but none in `FlowRenderer`.

Is this part accurate? What about an example? 🙂

Reply:
I copied this as-is from the old README, should have verified.